beacon-chain/core/blocks/withdrawals.go

```diff
@@ -162,6 +165,11 @@ func ProcessWithdrawals(st state.BeaconState, executionData interfaces.Execution
 		if err != nil {
 			return nil, errors.Wrap(err, "could not get withdrawals")
 		}
+
+		if len(wds) != len(expectedWithdrawals) {
+			return nil, fmt.Errorf("execution payload header has %d withdrawals when %d were expected", len(wds), len(expectedWithdrawals))
+		}
```

Comment:
Discussed offline with Potuz. This check isn't technically required since the withdrawal roots won't match if the length of withdrawals are different. However, I found it very helpful in debugging and it is essentially free to access the length of the slices.